src/streamlink/plugins/canalplus.py

```diff
@@ -55,15 +67,22 @@ def _get_streams(self):
                 return
             video_id = match.group('video_id')
 
-        res = http.get(self.API_URL.format(self.CHANNEL_MAP[channel], video_id))
-        videos = http.json(res, schema=self._api_schema)
+        res = http.get(self.API_URL.format(video_id))
+        try:
+            videos_list = http.json(res, schema=self.array_schema)
+            for vid in videos_list:
+                if vid['ID'] == video_id:
+                    videos = vid
+        except Exception:
+            videos = http.json(res, schema=self.object_schema)
         parsed = []
         headers = {'User-Agent': self._user_agent}
 
         # Some videos may be also available on Dailymotion (especially on CNews)
-        if videos['ID_DM'] != '':
-            for stream in self.session.streams('https://www.dailymotion.com/video/' + videos['ID_DM']).items():
-                yield stream
+        # note : disable dailymotion until plugin is fixed
```

Comment:
**dailymotion works fine** with an id, you don't need to disable this.

This and some other stuff will remove support for
- http://www.cnews.fr/france/video/accident-de-millas-lenquete-sur-le-passage-a-niveau-182909
- http://www.cnews.fr/direct
but it works without an issue on the normal streamlink version.